R/oauth-flow-auth-code.R

```diff
@@ -133,18 +146,35 @@ req_oauth_auth_code <- function(req, client,
 #'   token <- oauth_flow_auth_code(client, auth_url = "https://github.com/login/oauth/authorize")
 #'   token
 #' }
+#' @importFrom lifecycle deprecated
 oauth_flow_auth_code <- function(client,
                                  auth_url,
                                  scope = NULL,
                                  pkce = TRUE,
                                  auth_params = list(),
                                  token_params = list(),
-                                 host_name = "localhost",
+                                 host_name = deprecated(),
                                  host_ip = "127.0.0.1",
-                                 port = httpuv::randomPort()
+                                 port = httpuv::randomPort(),
+                                 redirect_uri = "http://localhost"
```

Comment:
I wonder if we should deprecate `port` too, replacing this default with something that automatically adds the random port.

Reply:
Oh that's what I said above, previously 😆. Let's leave this as is for this PR and consider a refactoring in a future PR: #291.